pkg/gui/controllers/helpers/refs_helper.go

```diff
@@ -271,6 +272,57 @@ func (self *RefsHelper) CreateGitResetMenu(ref string) error {
 	})
 }
 
+func (self *RefsHelper) CreateCheckoutMenu(commit *models.Commit) error {
+	branches := lo.Filter(self.c.Model().Branches, func(branch *models.Branch, _ int) bool {
+		return commit.Hash == branch.CommitHash && branch.Name != self.c.Model().CheckedOutBranch
+	})
+
+	ref := commit.Hash
+	var menuItems []*types.MenuItem
+
+	menuItems = append(menuItems, &types.MenuItem{
+		LabelColumns: []string{fmt.Sprintf("%s %s", self.c.Tr.CheckoutCommit, self.c.Tr.AsDetachedHead), style.FgRed.Sprint(ref[:8])},
+		OnPress: func() error {
+			self.c.Confirm(types.ConfirmOpts{
+				Title:  self.c.Tr.CheckoutCommit,
+				Prompt: self.c.Tr.SureCheckoutThisCommit,
+				HandleConfirm: func() error {
+					self.c.LogAction(self.c.Tr.Actions.CheckoutCommit)
+					return self.CheckoutRef(ref, types.CheckoutRefOptions{})
+				},
+			})
+			return nil
+		},
+		Tooltip: self.c.Tr.CheckoutCommitTooltip,
+		Key:     'd',
+	})
+
+	if len(branches) > 0 {
+		menuItems = append(menuItems, lo.Map(branches, func(branch *models.Branch, index int) *types.MenuItem {
+			return &types.MenuItem{
+				LabelColumns: []string{self.c.Tr.Actions.CheckoutBranch, style.FgBlue.Sprint(branch.Name)},
+				OnPress: func() error {
+					self.c.LogAction(self.c.Tr.Actions.CheckoutBranch)
+					return self.CheckoutRef(branch.RefName(), types.CheckoutRefOptions{})
+				},
+				Tooltip: self.c.Tr.CheckoutBranchTooltip,
+				Key:     rune(index + 1 + '0'), // Convert 1-based index to key
```

Comment:
This probably doesn't work very well when there are more than 9 branches. Maybe we don't need to worry about that because it will never happen for anybody in practice; but I still wonder if we should write the extra code to stop assigning Keys after the 9th branch.

Reply:
Good point. Stupid question from a non-go programmer: do we need to make the condition outside of the menu item or is it possible to assign an "empty" key to the `Key` attribute of MenuItem? The alternative is a bit verbosive:
```golang
if index < 9 {
  return &types.MenuItem {
    // ... the other properties
    Key: rune(index + 1 + '0'),
  }
} else {
  return &types.MenuItem {
    // ... the other properties as before
    // but now without the "Key" property
  }
}
```

Reply:
I added a fixup for this: 00a1aa8ea486